### PR TITLE
Detail view

### PR DESCRIPTION
**Pull request by mathieupost**
Move your mouse over a node!
Much detail!

### CODE REVIEW

File: src/main/java/tudelft/ti2806/pl3/detailView/DetailView.java [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+		Dimension size = getPreferredSize();
+		setBounds(x, y, size.width, size.height);
+	}
+
+	private void addLabels(String title, Set set) {
+		addLabels(title, set, null);
+	}
+
+	private void addLabels(String title, Set set, String filter) {
```
Comment (TomBrouws) [inline src/main/java/tudelft/ti2806/pl3/detailView/DetailView.java]: Missing JavaDoc!

**Comment by TomBrouws** [comment]
Sometimes duplicate Gene labels are displayed on a node.

File: src/main/java/tudelft/ti2806/pl3/data/wrapper/FixWrapper.java [hunk 2]
```diff
@@ -2,15 +2,21 @@
...
+	public static int ID_COUNT = -1;
+	public final int id;
 	public static final String ID_STRING = "[FIX]";
 	private Set<Genome> genome;
-	
+
+	public FixWrapper() {
+		id = ID_COUNT--;
+	}
+
```
Comment (Delfil) [inline src/main/java/tudelft/ti2806/pl3/data/wrapper/FixWrapper.java]: Why not ask an id in the constructor?

**Comment by Delfil** [comment]
This branch fixes the variable graph width issue & adds genome & label display.
Both work. Code looks fine. Can be merged.

File: src/main/java/tudelft/ti2806/pl3/controls/MouseManager.java [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+		super.init(graph, view);
+		view.setLayout(null);
+	}
+
+	@Override
+	public void mouseClicked(MouseEvent e) {
+		super.mouseClicked(e);
+		int x = e.getX();
+		int y = e.getY();
+		ArrayList<GraphicElement> graphicElements = view.allNodesOrSpritesIn(x - 5, y - 5, x + 5, y + 5);
```
Comment (Delfil) [inline src/main/java/tudelft/ti2806/pl3/controls/MouseManager.java]: Magic?

File: src/main/java/tudelft/ti2806/pl3/detailView/DetailView.java [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+ */
+public class DetailView extends JPanel {
+
+	/**
+	 * Constructs a DetailView and set a border and the preferred layout.
+	 */
+	public DetailView() {
+		BoxLayout layout = new BoxLayout(this, BoxLayout.Y_AXIS);
+		setLayout(layout);
+		setBorder(BorderFactory.createEmptyBorder(10, 10, 10, 10));
```
Comment (Delfil) [inline src/main/java/tudelft/ti2806/pl3/detailView/DetailView.java]: Magic

File: src/main/java/tudelft/ti2806/pl3/detailView/DetailView.java [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+			String nextString = iterator.next();
+			if (filter != null && filter.equals(nextString)) {
+				filter = null;
+				if (i >= 5) {
+					add(nextString);
+					i++;
+					break;
+				}
+			}
+			if (i < 5) {
```
Comment (Delfil) [inline src/main/java/tudelft/ti2806/pl3/detailView/DetailView.java]: Magic

File: src/main/java/tudelft/ti2806/pl3/data/wrapper/CombineWrapper.java [hunk 6]
```diff
@@ -66,7 +70,18 @@ public String getIdString() {
 	public int getId() {
 		return nodeList.get(0).getId();
 	}
-	
+
+	@Override
+	public Set<Label> getLabels() {
+		if (labels == null) {
+			labels = new HashSet<>();
```
Comment (borismattijssen) [inline src/main/java/tudelft/ti2806/pl3/data/wrapper/CombineWrapper.java]: You are creating a lot of HashSets this way. Have a look at getDataNodes and collectDataNodes for a more elegant solution.

File: src/main/java/tudelft/ti2806/pl3/data/wrapper/FixWrapper.java (hunk 2 above)
Comment (borismattijssen) [inline src/main/java/tudelft/ti2806/pl3/data/wrapper/FixWrapper.java]: Good point

File: src/main/java/tudelft/ti2806/pl3/controls/MouseManager.java [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+		for (GraphicElement graphicElement : graphicElements) {
+			System.out.println(graphicElement);
```
Comment (borismattijssen) [inline src/main/java/tudelft/ti2806/pl3/controls/MouseManager.java]: sysout? :P

File: src/main/java/tudelft/ti2806/pl3/controls/MouseManager.java [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+import java.awt.BorderLayout;
+import java.awt.event.MouseEvent;
+import java.util.ArrayList;
+
+/**
+ * This MouseManager extends the functionality of the default mouse actions of GraphStream.
+ * Created by Mathieu Post on 8-6-15.
+ */
+public class MouseManager extends DefaultMouseManager {
+	WrapperClone node = null;
```
Comment (borismattijssen) [inline src/main/java/tudelft/ti2806/pl3/controls/MouseManager.java]: maybe refactor into "previousWrapper" or something?

**Comment by borismattijssen** [comment]
Look at the comments. The rest looks fine!

File: src/main/java/tudelft/ti2806/pl3/controls/MouseManager.java (hunk 8 above)
Comment (mathieupost) [inline src/main/java/tudelft/ti2806/pl3/controls/MouseManager.java]: It is always the current node and if the mouse is currently not on a node it is set to null. So it is actually never the 'previous' node.

**Pull request merged**